syntax = "proto2";
option optimize_for = LITE_RUNTIME;
package ipfs.ipns;

enum KeyType {
  RSA = 0;
  Ed25519 = 1;
  Secp256k1 = 2;
  ECDSA = 3;
}

// PublicKey
message PublicKey {
  required KeyType Type = 1;
  required bytes Data = 2;
}

// PrivateKey
message PrivateKey {
  required KeyType Type = 1;
  required bytes Data = 2;
}
